server/pulp/plugins/util/publish_step.py

```diff
@@ -45,6 +45,20 @@ def _post_order(step):
         yield v
     yield step
 
+def _safe_makedirs(*args):
+    """
+    A wrapper for os.makedirs that handles the race condition of two directories being
+    created at the same time. An OSError is thrown if a file exists with the same name.
+
+    :param args: path[, mode] that goes to os.makedirs
+    :return:
+    """
+    try:
+        os.makedirs(*args)
+    except OSError:
+        if not os.path.isdir(args[0]):
```

Comment:
The newer style of exception handling is preferred. Python provides errno module that has all the error codes. The codes can vary from platform to platform. So it's better to use the module when checking IO errors and such. 

`except OSError as e:
    if e.errno != errno.EEXIST:
        raise`